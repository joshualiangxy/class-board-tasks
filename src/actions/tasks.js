import firebase from '../firebase/firebase';

export const addPersonalTask = task => ({ type: 'ADD_PERSONAL_TASK', task });

export const startAddPersonalTask = task => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;
    const id = task.id;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .set(task)
      .then(() => dispatch(addPersonalTask(task)));
  };
};

export const removePersonalTask = id => ({ type: 'REMOVE_PERSONAL_TASK', id });

export const startRemovePersonalTask = id => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .delete()
      .then(() => dispatch(removePersonalTask(id)));
  };
};

export const editPersonalTask = (id, updates) => ({
  type: 'EDIT_PERSONAL_TASK',
  id,
  updates
});

export const startEditPersonalTask = (id, updates) => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .set(updates)
      .then(dispatch(editPersonalTask(id, updates)));
  };
};

export const setTasks = tasks => ({ type: 'SET_TASK', tasks });

export const startSetTasks = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;
    const tasks = [];
    const userRef = firebase.firestore().collection('users').doc(uid);
    const tasksRef = userRef.collection('tasks');

    return tasksRef
      .get()
      .then(querySnapshot =>
        querySnapshot.forEach(queryDocSnapshot =>
          tasks.push({ id: queryDocSnapshot.id, ...queryDocSnapshot.data() })
        )
      )
      .then(() => {
        userRef.get().then(userDocSnapshot => {
          const groupIds = userDocSnapshot.get('groups') || [];
          const groupsPromises = [];

          groupIds.forEach(gid => {
            const groupPromises = [];
            const groupRef = firebase.firestore().collection('groups').doc(gid);

            groupPromises.push(
              groupRef
                .get()
                .then(groupDocSnapshot => groupDocSnapshot.get('name'))
            );
            groupPromises.push(groupRef.collection('tasks').get());

            groupsPromises.push(
              Promise.all(groupPromises).then(
                ([groupName, queryTaskCollectionSnapshot]) =>
                  queryTaskCollectionSnapshot.forEach(queryTaskDocSnapshot =>
                    tasks.push({
                      groupName,
                      gid,
                      id: queryTaskDocSnapshot.id,
                      ...queryTaskDocSnapshot.data()
                    })
                  )
              )
            );

            return Promise.all(groupPromises);
          });
        });
      })
      .then(() => dispatch(setTasks(tasks)));
  };
};

export const toggleCompletedPersonal = (id, completedState) => ({
  type: 'TOGGLE_COMPLETED_PERSONAL',
  id,
  completedState
});

export const startToggleCompletedPersonal = (id, completedState) => {
  return (dispatch, getState) => {
    const uid = getState().auth.user.uid;

    return firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .collection('tasks')
      .doc(id)
      .update({ completed: !completedState })
      .then(() => toggleCompletedPersonal(id, completedState));
  };
};

export const removeTaskData = () => ({ type: 'REMOVE_TASK_DATA' });
