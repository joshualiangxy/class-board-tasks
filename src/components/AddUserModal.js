import React, { useState } from 'react';
import Modal from 'react-modal';
import firebase from '../firebase/firebase';

// Note: This still works off uid for now. This also does NOT work yet.
const AddUserModal = ({ isOpen, onRequestClose, group }) => {
  const addNewUser = (userEmail, group) => {
    const uid = userEmail; // TODO: Change this when email-uid are linked!

    // Both of these references are DocumentReference
    const userRef = firebase.firestore().collection('users').doc(uid);
    const groupRef = firebase.firestore().collection('groups').doc(group);

    // Gets the data from a user.
    userRef.get().then(doc => {
      const data = doc.data();
      // Write displayName, studentNum, and UID into a new document
      if (typeof data === 'undefined') {
        // Most likely a user that doesn't exist.
        setError('No such user found');
      } else {
        groupRef
          .collection('users')
          .doc(uid) // Create document for this user's UID
          .set({
            studentNum: data.studentNum,
            displayName: data.displayName,
            uid: uid,
            admin: false
          })
          .then(() => {
            setError('User added!');
          })
          .catch(error => {
            setError('Something went wrong.');
          });
      }
      // Defaults to non-admin.
    });
  };

  // TODO: When calling this modal, pass in the group UID as a prop
  const [userEmail, setUserEmail] = useState('');
  const [error, setError] = useState('');
  // const [usersAdded, setUsersAdded] = useState([]); // For listing down people that have been added?

  const onUserEmailChange = e => {
    setUserEmail(e.target.value);
  };

  const onCancel = () => {
    setUserEmail('');
    onRequestClose();
  };

  const onSubmit = e => {
    e.preventDefault();
    const submittedEmail = userEmail.trim();
    if (!submittedEmail) {
      setError('Please enter a group name');
    } else {
      addNewUser(submittedEmail, group);
      // TODO: Catch user not found
      // Does not close after submitting, so that the user can continue to add more users

      // setUserEmail(''); // Also resets the field.
      // setError('Added!');
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      contentLabel="New Group"
      onRequestClose={() => onRequestClose()}
      appElement={document.getElementById('root')}
    >
      <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="User ID (required)"
          value={userEmail}
          onChange={onUserEmailChange}
          autoFocus
        />
        <button>Submit</button>
        <button onClick={onCancel}>Cancel</button>
        <div>{error}</div>
      </form>
    </Modal>
  );
};

export default AddUserModal;
