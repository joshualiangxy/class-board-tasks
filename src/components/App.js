import React from 'react';
import '../styles/App.css';

const App = () => (
  <div>
    <h1>This is my app</h1>
  </div>
);

export default App;

