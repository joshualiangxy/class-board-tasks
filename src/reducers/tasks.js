const tasksReducer = (state = [], action) => {
  switch (action.type) {
    case 'ADD_TASK':
      return [...state, action.task];
    case 'REMOVE_TASK':
      return state.filter(task => task.id !== action.id);
    case 'EDIT_TASK':
      return state.map(task =>
        task.id === action.id
          ? {
              ...task,
              ...action.updates
            }
          : task
      );
    case 'SET_TASK':
      return action.tasks;
    default:
      return state;
  }
};

export default tasksReducer;
