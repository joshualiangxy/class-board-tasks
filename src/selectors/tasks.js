const getSortedTasks = (tasks, { text, sortBy, grouped }) => {
  const lowercaseText = text.toLowerCase();

  const processedTasks = tasks
    .filter(
      task =>
        task.title.toLowerCase().includes(lowercaseText) ||
        task.description.toLowerCase().includes(lowercaseText) ||
        task.module.toLowerCase().includes(lowercaseText)
    )
    .sort((taskOne, taskTwo) => {
      switch (sortBy) {
        case 'deadline':
          if (taskOne.deadline && taskTwo.deadline)
            return taskOne.deadline - taskTwo.deadline;
          else if (taskOne.deadline) return -1;
          else if (taskTwo.deadline) return 1;
          else return taskOne.title.localeCompare(taskTwo.title);
        case 'name':
          return taskOne.title.localeCompare(taskTwo.title);
      }
    });

  const sortedList = {};

  if (grouped) {
    processedTasks.forEach(task => {
      if (task.groupName) {
        const gid = task.gid;

        if (!sortedList[gid])
          sortedList[gid] = { groupName: task.groupName, tasks: [] };
        sortedList[gid].tasks.push(task);
      } else if (task.module) {
        const module = task.module;

        if (!sortedList[module])
          sortedList[module] = { groupName: task.module, tasks: [] };
        sortedList[module].tasks.push(task);
      } else {
        if (!sortedList.others)
          sortedList.others = { groupName: 'others', tasks: [] };
        sortedList.others.tasks.push(task);
      }
    });
  } else sortedList.others = { groupName: 'others', tasks: processedTasks };

  return sortedList;
};

export default getSortedTasks;
