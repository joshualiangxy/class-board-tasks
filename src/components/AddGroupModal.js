import React, { useState } from 'react';
import Modal from 'react-modal';
import { startNewGroup } from '../actions/groups';
import { connect } from 'react-redux';

// TODO: Add the ability to add other users into the group
const AddGroupModal = ({ isOpen, onRequestClose, startNewGroup }) => {
  const [groupName, setGroupName] = useState('');
  const [module, setModule] = useState('');
  const [error, setError] = useState('');

  const onGroupNameChange = e => setGroupName(e.target.value);

  const onModuleChange = e => {
    const module = e.target.value.toUpperCase();

    setModule(module);
  };

  const onCancel = () => {
    setError('');
    setGroupName('');
    setModule('');
    onRequestClose();
  };

  const onSubmit = e => {
    e.preventDefault();
    const submittedName = groupName.trim();
    const submittedMod = module.trim();
    if (!submittedName) {
      setError('Please enter a group name');
    } else {
      setError('');
      setGroupName('');
      setModule('');
      startNewGroup(submittedName, submittedMod).then(() => onRequestClose());
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      contentLabel="New Group"
      onRequestClose={onCancel}
      closeTimeoutMS={200}
      appElement={document.getElementById('root')}
    >
      <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="Group name (required)"
          value={groupName}
          onChange={onGroupNameChange}
          autoFocus
        />
        <input
          type="text"
          placeholder="Module code/Class name (Optional)"
          value={module}
          onChange={onModuleChange}
        />
        <button>Submit</button>
        <button onClick={onCancel}>Cancel</button>
        <div>{error}</div>
      </form>
    </Modal>
  );
};

const mapDispatchToProps = dispatch => ({
  startNewGroup: (groupName, module) =>
    dispatch(startNewGroup(groupName, module))
});

export default connect(null, mapDispatchToProps)(AddGroupModal);
