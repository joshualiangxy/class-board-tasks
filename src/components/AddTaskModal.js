import React from 'react';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import { startAddPersonalTask, startAddGroupTask } from '../actions/tasks';
import TaskForm from './TaskForm';

export const AddTaskModal = ({
  startAddPersonalTask,
  startAddGroupTask,
  removeAdmin,
  isOpen,
  onRequestClose,
  gid,
  groupModule,
  groupName,
  addGroupTask,
  users
}) => {
  const submitTask = task =>
    gid
      ? startAddGroupTask(task, groupName)
          .then(isAdmin => {
            if (isAdmin) return addGroupTask(task);
            else return removeAdmin();
          })
          .then(() => onRequestClose())
      : startAddPersonalTask(task).then(() => onRequestClose());

  return (
    <Modal
      className="task-modal"
      isOpen={isOpen}
      contentLabel="Add Task"
      onRequestClose={onRequestClose}
      closeTimeoutMS={200}
      appElement={document.getElementById('root')}
    >
      <h1>Add Task</h1>
      <TaskForm
        gid={gid}
        groupModule={groupModule}
        users={users}
        submitTask={submitTask}
        onRequestClose={onRequestClose}
      />
    </Modal>
  );
};

const mapDispatchToProps = dispatch => ({
  startAddPersonalTask: task => dispatch(startAddPersonalTask(task)),
  startAddGroupTask: (task, groupName) =>
    dispatch(startAddGroupTask(task, groupName))
});

export default connect(undefined, mapDispatchToProps)(AddTaskModal);
