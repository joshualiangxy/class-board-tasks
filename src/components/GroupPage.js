import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import LoadingPage from './LoadingPage';
import GroupSettingsModal from './GroupSettingsModal';
import LeaveGroupModal from './LeaveGroupModal';
import AddTaskModal from './AddTaskModal';
import GroupTaskList from './GroupTaskList';
import { getAllUsers, leaveGroup } from '../actions/groups';
import { getAllGroupTasks } from '../actions/tasks';
import { history } from '../routers/AppRouter';

const GroupPage = ({
  gid,
  groupName,
  groupModule,
  authorised,
  uid,
  getAllUsers,
  getAllGroupTasks,
  leaveGroup
}) => {
  // For Modal
  const [addUserOpen, setAddUserOpen] = useState(false);
  const [leaveOpen, setLeaveOpen] = useState(false);
  const [addTaskOpen, setAddTaskOpen] = useState(false);
  const [users, setUsers] = useState([]);
  const [tasks, setTasks] = useState([]);
  const [admin, setAdmin] = useState(false);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    if (!authorised) history.push('/groups');
    else {
      const promises = [];
      promises.push(
        getAllUsers(gid).then(users => {
          users.sort((userOne, userTwo) =>
            userOne.displayName.localeCompare(userTwo.displayName)
          );
          setUsers(users);
        })
      );
      promises.push(getAllGroupTasks(gid).then(tasks => setTasks(tasks)));
      Promise.all(promises).then(() => setLoading(false));
    }
  }, [gid, authorised, getAllUsers, getAllGroupTasks]);

  useEffect(() => {
    if (users.length > 0) {
      const user = users.find(user => user.uid === uid);
      if (!user) {
        history.push('/groups');
        leaveGroup();
      } else setAdmin(user.admin);
    }
  }, [users, uid, leaveGroup]);

  const addGroupTask = task => setTasks([...tasks, task]);

  const editGroupTask = (id, updates) =>
    setTasks(tasks.map(task => (task.id === id ? updates : task)));

  const removeGroupTask = id => setTasks(tasks.filter(task => task.id !== id));

  const toggleGroupTaskComplete = id => {
    setTasks(
      tasks.map(task => {
        if (task.id === id) task.completed[uid] = !task.completed[uid];
        return task;
      })
    );
  };

  const removeUserFromTask = (id, uid) => {
    setTasks(
      tasks.map(task => {
        const newTask = { ...task };
        if (task.id === id) {
          const { [uid]: omit, ...rest } = task.completed;
          newTask.completed = rest;
        }
        if (task.uploadRequired && task.downloadURLs.hasOwnProperty(uid)) {
          const { [uid]: omit, ...rest } = task.downloadURLs;
          newTask.downloadURLs = rest;
        }

        return newTask;
      })
    );
  };

  const kickUserLocal = uid =>
    setTasks(
      tasks.map(task => {
        const newTask = { ...task };

        if (task.completed.hasOwnProperty(uid)) {
          const { [uid]: omit, ...rest } = task.completed;
          newTask.completed = rest;
        }
        if (task.uploadRequired && task.downloadURLs.hasOwnProperty(uid)) {
          const { [uid]: omit, ...rest } = task.downloadURLs;
          newTask.downloadURLs = rest;
        }

        return newTask;
      })
    );

  const closeAddUser = () => setAddUserOpen(false);

  const openAddUser = () => setAddUserOpen(true);

  const onLeaveCancel = () => setLeaveOpen(false);

  const openLeave = () => setLeaveOpen(true);

  const openAddTask = () => setAddTaskOpen(true);

  const closeAddTask = () => setAddTaskOpen(false);

  const removeAdmin = () => setAdmin(false);

  const renderLoad = () => setLoading(true);

  return (
    <div>
      {loading ? (
        <LoadingPage />
      ) : (
        <div>
          <h1>{groupName}</h1>
          {admin && (
            <div>
              <button onClick={openAddTask}>Add Task</button>
            </div>
          )}
          <button onClick={openAddUser}>Group Settings</button>
          <button onClick={openLeave}>Leave Group</button>
          <GroupTaskList
            removeAdmin={removeAdmin}
            removeUserFromTask={removeUserFromTask}
            tasks={tasks}
            users={users}
            admin={admin}
            groupName={groupName}
            editGroupTask={editGroupTask}
            removeGroupTask={removeGroupTask}
            toggleGroupTaskComplete={toggleGroupTaskComplete}
          />
          <GroupSettingsModal
            isOpen={addUserOpen}
            onRequestClose={closeAddUser}
            group={gid}
            users={users}
            setUsers={setUsers}
            admin={admin}
            setAdmin={setAdmin}
            uid={uid}
            kickUserLocal={kickUserLocal}
          />
          <LeaveGroupModal
            isOpen={leaveOpen}
            onRequestClose={onLeaveCancel}
            gid={gid}
            users={users}
            renderLoad={renderLoad}
            uid={uid}
            admin={admin}
          />
          <AddTaskModal
            removeAdmin={removeAdmin}
            isOpen={addTaskOpen}
            onRequestClose={closeAddTask}
            gid={gid}
            groupModule={groupModule}
            groupName={groupName}
            addGroupTask={addGroupTask}
            users={users}
          />
        </div>
      )}
    </div>
  );
};

const mapStateToProps = (state, { match }) => {
  const gid = match.params.id;
  const group = state.groups.find(group => group.gid === gid);
  const authorised = !!state.user.groups.find(id => id === gid);

  return {
    gid,
    groupName: group ? group.name : '',
    groupModule: group ? group.module : '',
    authorised,
    uid: state.auth.user.uid
  };
};

const mapDispatchToProps = (dispatch, { match }) => {
  const gid = match.params.id;

  return {
    getAllUsers: () => dispatch(getAllUsers(gid)),
    getAllGroupTasks: () => dispatch(getAllGroupTasks(gid)),
    leaveGroup: () => dispatch(leaveGroup(gid))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(GroupPage);
