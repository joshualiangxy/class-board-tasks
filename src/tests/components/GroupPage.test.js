import React from 'react';
import { shallow } from 'enzyme';
import {GroupPage} from '../../components/GroupPage';
import groupUsers from '../fixtures/groupUsers';
import { groupTasks } from '../fixtures/tasks';
import {browserHistory} from 'react-router'

jest.mock('react-router', () => ({
  browserHistory: {
    push: jest.fn()
  }
}))

const gid = 'one';
const groupName = 'nameOne';
const groupModule = 'moduleOne';
const authorised = true;
const uid = 'testuid';
const getAllUsers = jest.fn(() => Promise.resolve(groupUsers));
const getAllGroupTasks = jest.fn(() => Promise.resolve(groupTasks));
const leaveGroup = jest.fn();

const wrapper = shallow(
  <GroupPage
    gid={gid}
    groupName={groupName}
    groupModule={groupModule}
    authorised={authorised}
    uid={uid}
    getAllUsers={getAllUsers}
    getAllGroupTasks={getAllGroupTasks}
    leaveGroup={leaveGroup}
  />
);

describe('render', () => {
  it('should render if the user is part of the group', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should push if the user is not part of the group', () => {
    wrapper.setProps({uid: 'nothere'})
    // expect(browserHistory.push).toHaveBeenCalledTimes(1);
  })
});
