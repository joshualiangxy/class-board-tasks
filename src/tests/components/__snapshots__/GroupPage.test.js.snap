// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render if the user is part of the group 1`] = `
<div>
  <div>
    <div
      className="page-header"
    >
      <div
        className="content-container"
      >
        <h1
          className="page-header__title"
        >
          nameOne
        </h1>
        <div
          className="page-header__buttons"
        >
          <button
            className="button button--norm page-header__actions"
            onClick={[Function]}
          >
            Add Task
          </button>
          <div>
            <button
              className="button button--norm page-header__actions button--right"
              onClick={[Function]}
            >
              Group Settings
            </button>
            <button
              className="button button--norm page-header__actions"
              onClick={[Function]}
            >
              Leave Group
            </button>
          </div>
        </div>
         
      </div>
    </div>
    <GroupTaskList
      admin={true}
      editGroupTask={[Function]}
      groupName="nameOne"
      removeAdmin={[Function]}
      removeGroupTask={[Function]}
      removeUserFromTask={[Function]}
      tasks={
        Array [
          Object {
            "completed": Object {
              "testuid": false,
            },
            "description": "Finish by Sunday!",
            "downloadURLs": Object {
              "testuid": Object {
                "downloadURL": "website.com",
                "fileName": "Homework.pdf",
              },
            },
            "gid": "72b6b3bb-cb74-4d08-ae67-dfc1b51baaf9",
            "id": "076e610e-aade-4d2b-897f-81a29f00c76b",
            "module": "CS1231",
            "title": "Homework 1",
            "uploadRequired": true,
          },
          Object {
            "completed": Object {},
            "deadline": 1593403200000,
            "description": "",
            "gid": "ce5c9b1b-e4d2-4dd2-bae4-7c559e8235ea",
            "id": "0b80e917-78a4-4f87-a379-cdc1733923d6",
            "module": "CS2040S",
            "title": "Do problem set",
          },
          Object {
            "completed": Object {
              "testuid": false,
            },
            "deadline": 1593489600000,
            "description": "Finish readings",
            "downloadURLs": Object {},
            "gid": "3e28d54d-ebc1-4469-be21-e59ce210f245",
            "id": "c1511c73-07b9-44cd-a4bd-01758515acb5",
            "module": "GEQ1000",
            "title": "Physics readings",
            "uploadRequired": true,
          },
        ]
      }
      toggleGroupTaskComplete={[Function]}
      users={
        Array [
          Object {
            "admin": false,
            "displayName": "Peter",
            "studentNum": "A0000000A",
            "uid": "testuid2",
          },
          Object {
            "admin": true,
            "displayName": "Robert",
            "studentNum": "A0123456A",
            "uid": "testuid",
          },
        ]
      }
    />
    <Connect(GroupSettingsModal)
      admin={true}
      group="one"
      isOpen={false}
      kickUserLocal={[Function]}
      onRequestClose={[Function]}
      setAdmin={[Function]}
      setUsers={[Function]}
      uid="testuid"
      users={
        Array [
          Object {
            "admin": false,
            "displayName": "Peter",
            "studentNum": "A0000000A",
            "uid": "testuid2",
          },
          Object {
            "admin": true,
            "displayName": "Robert",
            "studentNum": "A0123456A",
            "uid": "testuid",
          },
        ]
      }
    />
    <Connect(LeaveGroupModal)
      admin={true}
      gid="one"
      isOpen={false}
      onRequestClose={[Function]}
      renderLoad={[Function]}
      uid="testuid"
      users={
        Array [
          Object {
            "admin": false,
            "displayName": "Peter",
            "studentNum": "A0000000A",
            "uid": "testuid2",
          },
          Object {
            "admin": true,
            "displayName": "Robert",
            "studentNum": "A0123456A",
            "uid": "testuid",
          },
        ]
      }
    />
    <Connect(AddTaskModal)
      addGroupTask={[Function]}
      gid="one"
      groupModule="moduleOne"
      groupName="nameOne"
      isOpen={false}
      onRequestClose={[Function]}
      removeAdmin={[Function]}
      users={
        Array [
          Object {
            "admin": false,
            "displayName": "Peter",
            "studentNum": "A0000000A",
            "uid": "testuid2",
          },
          Object {
            "admin": true,
            "displayName": "Robert",
            "studentNum": "A0123456A",
            "uid": "testuid",
          },
        ]
      }
    />
  </div>
</div>
`;
